src/sentry/migrations/0378_backfill_perf_issue_events_issue_platform.py

```diff
@@ -0,0 +1,648 @@
+        while next_offset is not None:
+            rows, next_offset = _query_performance_issue_events(
+                project_id=project_id,
+                start=START_DATETIME,
+                end=END_DATETIME,
+                offset=next_offset,
+            )
+
+            for row in rows:
+                group_id = row["group_id"]
+                event_id = row["event_id"]
+                finish_ts = row["finish_ts"]
+
+                # avoid re-ingesting the same transaction since we'll be dual-writing
+                if event_id not in already_processed_events[group_id]:
+                    group_progress: ProjectGroupTotalProgress = previous_progress.get_or_init(
+                        group_id
+                    )
+                    error_maybe = group_progress.consume_error()
+                    if error_maybe is None:
+                        already_processed = group_progress.already_processed(finish_ts, event_id)
+                        if already_processed:
+                            continue
+                        else:
+                            backfill_perf_issue_occurrence(
+                                row,
+                                Group,
+                                GroupHash,
+                                print_success
+                                if not WRITE_TO_FILE
+                                else group_progress.save_success_to_file,
+                                print_exception_on_error
+                                if not WRITE_TO_FILE
+                                else group_progress.save_error_to_file,
+                                dry_run=dry_run,
+                            )
+
+                    else:
+                        # resume progress from the (finish_ts, event_id) in the last processed error
+                        # last_error_finish_ts = error_maybe["finish_ts"]
+                        # last_error_event_id = error_maybe["event_id"]
+                        # last_error_exception = error_maybe["exception"]
+                        # last_error_nodestore_saved = error_maybe["nodestore_saved"]
+                        # last_error_eventstream_sent = error_maybe["eventstream_sent"]
+                        # TODO: implement some sort of pagination scheme where we pass back in the finish_ts
+                        pass
+
+
+def backfill_perf_issue_occurrence(
+    row: TransactionRow,
+    Group: Any,
+    GroupHash: Any,
+    on_success: BackfillEventSuccess = lambda *args: None,
+    on_exception: BackfillEventError = lambda *args: None,
+    dry_run: bool = True,
+) -> None:
+    occurrence_nodestore_saved = False
+    occurrence_eventstream_sent = False
+
+    try:
+        project_id = row["project_id"]
+        group_id = row["group_id"]
+        event_id = row["event_id"]
+        finish_ts = row["finish_ts"]
+
+        group = Group.objects.get(id=group_id)
+        group_hash = GroupHash.objects.get(group_id=group_id)
```

Comment:
It might be a good idea here to use `values_list` to just fetch the fields we need. This migration will already take a while, so small optimizations might help a fair bit